test/deploy.smart.wallet.test.ts

```diff
@@ -0,0 +1,64 @@
+import { RelayingServices, SmartWallet } from '../src';
+import { MockRelayingServices, Web3Mock } from './mock';
+import Expect = jest.Expect;
+import { MOCK_SMART_WALLET_ADDRESS, MOCK_TOKEN_ADDRESS } from './constants';
+
+declare const expect: Expect;
+
+describe('Tests for smart wallet', () => {
+    let sdk: RelayingServices;
+
+    beforeEach(async () => {
+        sdk = new MockRelayingServices(
+            new Web3Mock({
+                getCodeEmpty: false
+            }) as any
+        );
+        await sdk.initialize({});
+    });
+
+    it('Should fail with smart wallet already deployed', async () => {
+        try {
+            await sdk.deploySmartWallet(
+                {
+                    address: MOCK_SMART_WALLET_ADDRESS,
+                    index: 0,
+                    deployed: true
+                },
+                MOCK_TOKEN_ADDRESS,
+                0
+            );
+            fail('The smart wallet this not already deployed');
+        } catch (error: any) {
+            expect(error.message).toBe('Smart Wallet already deployed');
+        }
+    });
+});
+
+describe('Tests for smart wallet without being deployed', () => {
+    let sdk: RelayingServices;
+
+    beforeEach(async () => {
+        sdk = new MockRelayingServices(
+            new Web3Mock({
+                getCodeEmpty: true
+            }) as any
+        );
+        await sdk.initialize({});
+    });
+
+    it('Should deploy smart wallet successfully', async () => {
+        const smartWallet: SmartWallet = await sdk.deploySmartWallet(
+            {
+                address: MOCK_SMART_WALLET_ADDRESS,
+                index: 0,
+                deployed: true
+            },
+            MOCK_TOKEN_ADDRESS,
+            0
+        );
+        expect(smartWallet.address).toBe(MOCK_SMART_WALLET_ADDRESS);
+        expect(smartWallet.index).toBe(0);
+        expect(smartWallet.deployed).toBeTruthy();
```

Comment:
why have we removed the previous checks?
```
expect(smartWallet.tokenAddress).toBe(MOCK_TOKEN_ADDRESS);
expect(smartWallet.deployTransaction).toBe(MOCK_TRANSACTION_RECEIPT);
```
Aren't them required anymore? if so, why?

Reply:
The token address and the transaction receipt is not return from the smart wallet deploy, it is only added as mock in the first version of test.

Reply:
From the [code](https://github.com/wilsoniovlabs/relaying-services-sdk/blob/fixes/relaying-services-sdk/src/sdk.ts#L296), it seems that both `tokenAddress` and `deployTransaction` are returned from the `deploySmartWallet` function. Am I missing something?

Reply:
`const smartWalletAddress = await smartWalletFactory.methods
            .getSmartWalletAddress(
                this.getAccountAddress(),
                ZERO_ADDRESS,
                smartWalletIndex
            )
            .call();
`
This is the new return in this method.

Reply:
It is fixed.